app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -297,6 +297,34 @@ def restart_middleware_deployment(ems_ref, deployment_name)
       end
     end
 
+    def add_middleware_jdbc_driver(ems_ref, hash)
+      with_provider_connection do |connection|
+        driver_data = {
+          :driver_name          => hash[:driver]["driver_name"],
+          :driver_jar_name      => hash[:driver]["driver_jar_name"] || hash[:driver]["file"].original_filename,
+          :module_name          => hash[:driver]["module_name"],
+          :driver_class         => hash[:driver]["driver_class"],
+          :driver_major_version => hash[:driver]["driver_major_version"],
+          :driver_minor_version => hash[:driver]["driver_minor_version"],
+          :binary_content       => hash[:driver]["file"].read,
+          :resource_path        => ems_ref.to_s
+        }
+
+        actual_data = {}
+        connection.operations(true).add_jdbc_driver(driver_data) do |on|
+          on.success do |data|
+            _log.debug "Success on websocket-operation #{data}"
+            actual_data[:data] = data
+          end
+          on.failure do |error|
+            actual_data[:data] = {}
+            actual_data[:error] = error
+            _log.error 'error callback was called, reason: ' + error.to_s
+          end
+        end
```

Comment:
It's not necessary to store the result in `actual_data` if it's not being used anywhere.

```ruby
        connection.operations(true).add_jdbc_driver(driver_data) do |on|
           on.success do |data|
             _log.debug "Success on websocket-operation #{data}"
           end
           on.failure do |error|
             _log.error 'error callback was called, reason: ' + error.to_s
           end
         end
```

Should be enough.